src/wallet/apiHelpers.ts

```diff
@@ -31,38 +30,40 @@ export const getBlock = async (
 // TODO: handle pagination
 export const getTransactions = async (
   address: string,
-  apiEndpoint: string = API_ENDPOINT
+  apiEndpoint: string
 ): Promise<Api.TransactionsResponse> => {
-  const getTx = async (n: number, skip: number): Promise<Api.Transaction[]> => {
-    // eslint-disable-next-line
-    const response = await fetch(
-      `${apiEndpoint}/transactions/address/${address}?limit=${n}&skip=${skip}`,
-      {
-        mode: 'cors',
+  try {
```

Comment:
Not sure the try catch belongs here.  I think we want to bubble the error up instead of handling it silently here. What do you think?

Reply:
Agreed, but currently without it the app hangs on login when using the testnet because of the API errors. This was the easiest way to ensure that an empty wallet was generated in that case. I think keeping this here is just a short term solution

Reply:
Makes sense -- let's ensure that the wallet doesn't just hang. I'm trying to think of somewhere else to put this.

Reply:
Ideally moving from login screen to the wallet itself would not be dependent on anything remote, but just on decrypting the seed and confirming it's a valid one. That should get the user to the wallet interface, wherein they'd subsequently see any errors related to networking issues, etc... Does that make sense? It's ok if we can't get that worked out for the MVP, but longer term we should be aiming for that.

Reply:
To make sure we're all on the same page I'll clarify the flow:
- Wallet is initialized by decrypting the seed (or creating a new wallet)
- We pass the wallet instance to the `getWalletInfo()` method to save it to the state. During the call to this method we:
    - Set the network, if one is found in local storage
    - Call `Wallet.addressDiscovery()` (this uses the transactions API)
    - Now it is saved to the state and we are done in this method
- Next we redirect to the balance page
- When this page is first loaded, we call `updateState()` to fetch the latest balance and transactions. This is where it hangs if API calls fail and are not handled properly. So we are already on the wallet page, and just have no data to show

So, because we search for all addresses before redirecting to the balance page, the try/catch here is required to handle API failures when searching for transactions. I agree with you Ben that it shouldn't be a blocker to loading the app. I think it makes sense to work with Ed on the best way to do that in a separate PR, because right now when first logging in the `addressDiscovery()` and `updateStat()` calls are redundant anyway, so we can tie this together in one, new PR

However, I did have a bug where the API endpoint was not being set properly in the wallet class, so I just pushed a commit that fixes that (and as a very nice side-effect, also fixes the tests) 🙂

I also pushed a commit to improve error handling in the mounted hook. So everything will load as expected (just very slowly on this branch because of all the calls). And look, results from the testnet!  🎉 

![image](https://user-images.githubusercontent.com/17163988/88926814-5b92de80-d22b-11ea-9fb8-1a115c9535cb.png)

Reply:
Hey, let me chime in on the log-in flow. Similar to Matt's with minor revisions.

- log in - create a new wallet or restore one. Pass wallet state to getWalletInfo method. Set the network. Restore state from cache. Don't do address discovery yet.
- balance page - IF the user doesn't have stuff cached (new wallet or restored from seed/wallet file with no cache), start `addressDiscovery` in a try/catch. If the user does have stuff cached, then just call `updateState` in a try/catch. From now on, the state will be updated with an `updateState` call.

Reply:
That flow sounds much better! I can make that change this afternoon or feel free to do it earlier if you don’t mind